monitoring/perf_level.cc

```diff
@@ -4,25 +4,38 @@
 //  (found in the LICENSE.Apache file in the root directory).
 //
 
+#include "rocksdb/perf_level.h"
+
 #include <assert.h>
-#include "monitoring/perf_level_imp.h"
 
-namespace rocksdb {
+#include "monitoring/perf_level_by_bitfield_imp.h"
 
-#ifdef ROCKSDB_SUPPORT_THREAD_LOCAL
-__thread PerfLevel perf_level = kEnableCount;
-#else
-PerfLevel perf_level = kEnableCount;
-#endif
+namespace rocksdb {
 
 void SetPerfLevel(PerfLevel level) {
+  // TODO: discuss keep assertions or not
   assert(level > kUninitialized);
   assert(level < kOutOfBounds);
-  perf_level = level;
+  switch (level) {
+    case kEnableCount:
+      perf_bit_field = BitFieldEnableCount;
+      break;
+    case kEnableTimeExceptForMutex:
+      perf_bit_field = BitFieldEnableTimeExceptForMutex;
+      break;
+    case kEnableTimeAndCPUTimeExceptForMutex:
+      perf_bit_field = BitFieldEnableTimeAndCPUTimeExceptForMutex;
+      break;
+    case kEnableTime:
+      perf_bit_field = BitFieldEnableTime;
+      break;
+    default:
+      perf_bit_field = {};
+      break;
+  }
+  perf_bit_field.perf_level = level;
 }
 
-PerfLevel GetPerfLevel() {
-  return perf_level;
-}
+PerfLevel GetPerfLevel() { return (PerfLevel)perf_bit_field.perf_level; }
```

Comment:
What if we set a perf level, then assign by flags? This will produce wrong outputs.